src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -376,6 +420,25 @@ func isValidMacLogin(
 	return hmac.Equal(givenMac, expectedMAC), nil
 }
 
+// checkFlowsEqual checks if two registration flows have the same stages
+// within them. Order of stages does not matter.
+func checkFlowsEqual(aFlow, bFlow authtypes.Flow) bool {
+	a := aFlow.Stages
+	b := bFlow.Stages
+	if len(a) != len(b) {
```

Comment:
it's valid for the client to complete *extra* stages.

Reply:
This is still going to fail if the client completes extra stages.

You're also going to have to rename the function and change its comment.